crates/optimism/flashblocks/src/sequence.rs

```diff
@@ -259,171 +246,437 @@ impl Deref for FlashBlockCompleteSequence {
     }
 }
 
-impl<T> TryFrom<FlashBlockPendingSequence<T>> for FlashBlockCompleteSequence {
+impl TryFrom<FlashBlockPendingSequence> for FlashBlockCompleteSequence {
     type Error = eyre::Error;
-    fn try_from(sequence: FlashBlockPendingSequence<T>) -> Result<Self, Self::Error> {
-        Self::new(
-            sequence.inner.into_values().map(|block| block.block().clone()).collect::<Vec<_>>(),
-            sequence.execution_outcome,
-        )
+    fn try_from(sequence: FlashBlockPendingSequence) -> Result<Self, Self::Error> {
+        Self::new(sequence.inner.into_values().collect(), sequence.execution_outcome)
     }
 }
 
-#[derive(Debug)]
-struct PreparedFlashBlock<T> {
-    /// The prepared transactions, ready for execution
-    txs: Vec<WithEncoded<Recovered<T>>>,
-    /// The tracked flashblock
-    block: FlashBlock,
-}
```

Comment:
we should add this back because I believe rn we're re-recovering the entire block per flashblock in

https://github.com/paradigmxyz/reth/blob/main/crates/optimism/flashblocks/src/cache.rs#L177-L177

recovery is fairly expensive and we can avoid doing all of this for the previous flashblocks in the block